File: shared/messages/room-change-message.ts
import { MessageType } from "../enum/message-type";
import { BaseMessage } from "./base-message";

export class RoomChangeMessage extends BaseMessage {
  message_type = MessageType.ROOM_CHANGE_MESSAGE;
  room?: string;

  constructor(data: Partial<RoomChangeMessage>) {
    super();
    Object.assign(this, data);
  }
}

File: src/main/java/ubc/cosc322/messages/RoomChangeMessage.java
package ubc.cosc322.messages;

public class RoomChangeMessage extends BaseMessage {
  public String room = "";

  public RoomChangeMessage() {
    
  }

  public RoomChangeMessage(String room) {
    message_type = MessageType.ROOM_CHANGE_MESSAGE;
    this.room = room;
  }
}
